src/lib/libast/path/pathstat.c

```diff
@@ -24,15 +24,15 @@
 #include <errno.h>
 #include <sys/stat.h>
 
-/*
- * physical stat if logical fails
- */
+//
+// First try `stat()` and if that fails try `lstat()`
+//
+int pathstat(const char *path, struct stat *statbuf) {
+    int old_errno;
 
-int pathstat(const char *path, struct stat *st) {
-    int oerrno;
-
-    oerrno = errno;
-    if (!stat(path, st)) return 0;
-    errno = oerrno;
-    return lstat(path, st);
+    old_errno = errno;
+    if (!stat(path, statbuf)) return 0;
+    errno = old_errno;
```

Comment:
I'm not suggesting you remove this but for the life of me I can't see the point of restoring the original errno value. Any code using this function (and there is only one in the project as I type this) it should only look at errno if this function returns -1 to indicate an error. Thus there should be no need to save/restore errno.